src/Vector.hpp

```diff
@@ -345,243 +348,174 @@ Vector<T, N> pow(Vector<T, N> arg, EXP_T exponent) {
   return arg.pow(exponent);
 }
 
+/// unary `-` (minus) operator for Vector class
+template <typename T, int N> Vector<T, N> operator-(const Vector<T, N> &a) {
+  Vector<T, N> ret{};
+  std::transform(a.begin(), a.end(), ret.begin(), std::negate<T>());
+  return ret;
+}
+
 /*
-template<typename T1,typename T2,int N>
-bool operator ==(const Vector<T1,N> &arg1, const Vector<T2,N> &arg2) {
-    bool ret = true;
-    for (size_t i = 0; i < N; ++i) {
-        ret &= arg1[i] == arg2[i];
-    }
-    return ret;
+ * Binary operations.
+ *
+ * First four are Vector-Vector elementwise operations (V[i] op U[i] for all i).
+ * Second four are scalar-Vector elementwise operations (k op V[i] for all i).
+ * Third four are Vector-scalar elementwise operations (V[i] op k for all i).
+ */
+
+/// binary `+` (plus) operator for two Vectors
+template <typename T, int N>
+Vector<T, N> operator+(const Vector<T, N> &a, const Vector<T, N> &b) {
+  Vector<T, N> ret{};
```

Comment:
not sure we should zero-initialise here right before overwriting the zeros. I know it seems kinda trivial, but it doubles the cost of the op.